src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -3339,8 +3339,10 @@ internal static string GetDriveQualifiedPath(string path, PSDriveInfo drive)
                 }
                 else
                 {
-                    if (path.StartsWith(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) || 
-                        path.StartsWith(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)) 
+                    if ((path.StartsWith(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) || 
+                         path.StartsWith(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)) &&
+                        !(path.Equals(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) ||
+                         path.Equals(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)))
```

Comment:
Why not `path.IndexOfAny(new char[]{ '\\', '/' }, 0, 1) == 0`? It's concise and efficient -- only check the first character of `path` essentially.